src/Debugger/Debuggee.php

```diff
@@ -0,0 +1,285 @@
+     *            status.
+     *      @type ExtendedSourceContext[] $extSourceContexts References to the locations and
+     *            revisions of the source code used in the deployed application.
+     */
+    public function __construct(ConnectionInterface $connection, array $info = [])
+    {
+        $this->connection = $connection;
+
+        $info += [
+            'id' => null,
+            'isInactive' => false,
+            'agentVersion' => DebuggerClient::DEFAULT_AGENT_VERSION,
+            'status' => null,
+            'extSourceContexts' => [],
+            'uniquifier' => null,
+            'description' => null
+        ];
+
+        $this->id = $info['id'];
+        $this->project = $info['project'];
+        $this->uniquifier = $info['uniquifier'];
+        $this->description = $info['description'];
+        $this->status = $info['status'];
+        $this->agentVersion = $info['agentVersion'];
+        $this->isInactive = $info['isInactive'];
+        $this->extSourceContexts = $info['extSourceContexts'];
+    }
+
+    /**
+     * Return the debuggee identifier.
+     *
+     * Example:
+     * ```
+     * echo $debuggee->id();
+     * ```
+     *
+     * @return string
+     */
+    public function id()
+    {
+        return $this->id;
+    }
+
+    /**
+     * Register this debuggee with the Stackdriver backend.
+     *
+     * Example:
+     * ```
+     * $debuggee->register();
+     * echo $debuggee->id();
+     * ```
+     *
+     * @param array $args
+     * @return bool
+     */
+    public function register(array $args = [])
+    {
+        $resp = $this->connection->registerDebuggee(['debuggee' => $this] + $args);
+        if (array_key_exists('debuggee', $resp)) {
+            $this->id = $resp['debuggee']['id'];
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Fetch the list of breakpoints this debugee should try to handle.
+     *
+     * Example:
+     * ```
+     * $breakpoints = $debuggee->breakpoints();
+     * ```
+     *
+     * @param array $options [optional] {
+     *      Configuration options.
+     *
+     *      @type string $waitToken A wait token that, if specified, blocks the
+     *            method call until the list of active breakpoints has changed,
+     *            or a server selected timeout has expired. The value should be
+     *            set from the last returned response.
+     *      @type bool $successOnTimeout If set to **true**, returns
+     *            **google.rpc.Code.OK** status and sets the **waitExpired**
+     *            response field to **true** when the server-selected timeout
+     *            has expired (recommended). If set to **false**, returns
+     *            **google.rpc.Code.ABORTED** status when the server-selected
+     *            timeout has expired (deprecated).
+     * }
+     * @return Breakpoint[]
+     */
+    public function breakpoints(array $options = [])
```

Comment:
I see they are imported, should this method be utilizing the iterators?

Reply:
This API doesn't appear to have any kind of pagination. I guess we could still wrap it in an iterator.

Reply:
No need if that is the case, o was only curious because I saw the imports.